ngApp/controllers/controllers.ts

```diff
@@ -9,7 +9,93 @@ namespace codingskills.Controllers {
 
     }
     export class CourtsideController {
+        constructor(
+            private $http: ng.IHttpService,
+            private $state: ng.ui.IStateService
+            ) {
+                console.log('The state service is ', $state);
+                this.getWords();
+        }
+        public words = [];
+        public currentWord;
+        public typed;
+        public difference;
+        public i = 0;
+        public gameRunning = false;
+        public statsObject = {
+            mistakes: 0,
+            wordsTyped: 0,
+            keysTyped: 0,
+            errorRate: 0
+        }
+        public runGame() {
+            if(this.statsObject.wordsTyped == 0 && this.typed == 1 && !this.gameRunning) {
+                this.gameRunning = true;
+                console.log('game started? ', this.gameRunning);
+                let game = window.setTimeout(() => {
+                    console.log("I wanted to use the state obj", this.$state);
+                    this.$state.go('lockerroom', this.statsObject);
+                        }, 15000);
+            }
+        }
+        public getWords() {
+            let pattern = '[s]';
+            this.$http.get('https://wordsapiv1.p.mashape.com/words/?mashape-key=L1Q3tAzB6rmshe27MNaoQquiTyTVp1aw7icjsnz3QOFVipm7Bv&letterPattern='

```

Comment:
if you treat this endpoint as API in your server-side you can secure the key.   not a blocker.